Multimodal autoencoder (#145)
creatis-myriad/creatis-myriad.github.io: collections/_posts/2023-07-17-Mutlimodal-AutoEncoder.md

Reviewer: There's a typo in the name of the fle as well :sweat_smile: _Mutlimodal_ -> _Multimodal_

Author: I change the filename, thanks !

* Masks random patches of the input image and random tokens of the input text and reconstructs the missing pixels and tokens
> this makes pre-training a self-supervised process 

* Uses different masking rates for input images and text due to the different information densities of vision and language

&nbsp;

## Formalism

##### Loss function

$$\theta^{*},\theta_1^{*},\theta_2^{*}=\arg \min_{\theta,\theta_1,\theta_2} \sum_{s=1}^{2} L_s\left( Y_s,D_{\theta_s} \left( M_{\theta}(I,T) \right) \right)$$
	
- $$L_s$$ are the loss functions of pretext tasks, i.e MSE between the reconstructed and original images and the negative log-likelihood for the masked tokens

- $$D_{\theta_s}$$ are the decoders with their parameters $$\theta_1$$, $$\theta_2$$

- $$M_{\theta}$$ is the backbone model with its parameters $$\theta$$.

&nbsp;

##### Vision encoder

$$X^{\nu} \in \mathbb{R}^{(N+1) \times D} \,=\, \left[ p_I; p_1 E^{\nu}; \cdots; p_N E^{\nu} \right]\,+\,E^{\nu}_{pos}$$

- Each image $$I \in \mathbb{R}^{H \times W \times C}$$ is divided into $$N$$ patches $$\{ p_1,\cdots,p_N \}$$

- $$E^{\nu} \in \mathbb{R}^{P^2 \times D}$$ is the projection matrix into the patch embeddings

> - $$p_I \in \mathbb{R}^{D}$$ used for the aggregation of visual information

Reviewer: Missing word:
```suggestion
- $$p_I \in \mathbb{R}^{D}$$ is used for the aggregation of visual information
```

- $$X^{\nu}$$ is fed into a transformer model with $$N_{\nu}$$ transformer blocks to obtain the contextualized image representation $$H^{\nu} \in \mathbb{R}^{(N+1) \times D} \,=\, \left[ h^{\nu}_I; h^{\nu}_1; \cdots; h^{\nu}_N \right]$$

&nbsp;

##### Language encoder

$$X^{l} \in \mathbb{R}^{(M+2) \times D} \,=\, \left[ w_T; w_1 E^{l}; \cdots; w_M E^{l}; w_{SEP} \right]\,+\,E^{l}_{pos}$$

> - Each input text is tokenize to subword tokens $${w_1,\cdots;w_M}$$ by WordPiece, where tokens $$w_m \in \mathbb{R}^{V}$$ are represented in one-hot form and $$V$$ is the vocabulary size

Reviewer: Conjugation:
```suggestion
- Each input text is tokenized to subword tokens $${w_1,\cdots;w_M}$$ by WordPiece, where tokens $$w_m \in \mathbb{R}^{V}$$ are represented in one-hot form and $$V$$ is the vocabulary size
```

- $$E^{l} \in \mathbb{R}^{V \times D}$$ is the projection matrix into the text embeddings

- $$w_T \in \mathbb{R}^{D}$$ and $$w_{SEP} \in \mathbb{R}^{D}$$ correspond to a start-of-sequence token embedding and a special boundary token embedding, respectively 

- $$X^{l}$$ is fed into a transformer model with $$N_{l}$$ transformer blocks to obtain the contextualized text representation $$H^{l} \in \mathbb{R}^{(M+2) \times D} \,=\, \left[ h^{l}_T; h^{l}_1; \cdots; h^{l}_M; h^{l}_{SEP} \right]$$

&nbsp;

##### Masking scheme

- the authors used random sampling with a much greater masking ratio for images (i.e. $$75\%$$) than for texts (i.e. $$15\%$$). This is justified by the fact that images are redundant while languages are information-dense

&nbsp;

##### Representation selection for reconstruction

> - Images and texts are abstracted at different levels, with pixels having a lower semantic level than token of texts.

Reviewer: Phrasing:
```suggestion
- Images and texts are abstracted at different levels, with pixels having a lower semantic level than text tokens.
```

* Accuracy is used as metric for the Med-VQA and medical Image-Text classification tasks

* Recall@K (K=1, 5, 10) is used for the Medical Image-Caption Retrieval task

> Unfortunately, nothing is said concerning the fine-tuning of the pretrained methods for the different downstream tasks :(

&nbsp;

## Results for Med-VQA task

![](/collections/images/multimodal_autoencoder/med-VQA-results.jpg)

&nbsp;

## Results for Medical Image-Text Classification

![](/collections/images/multimodal_autoencoder/classification-results.jpg)

&nbsp;

## Results for Medical Image-Caption Retrieval

![](/collections/images/multimodal_autoencoder/retrieval-results.jpg)

(ZS) means zero-shot and (FT) means fine-tuning 

&nbsp;

## Ablation study

![](/collections/images/multimodal_autoencoder/ablation-study.jpg)

(MIM) stands for Masked Image Modeling and (MLM) stands for Masked Language Modeling

&nbsp;

## Qualitative results

![](/collections/images/multimodal_autoencoder/qualitative-results.jpg)

&nbsp;

# Conclusions

* Image/text coupling for medical data analysis looks like a promising way forward

> * Pre-training in a self-supervised way using maskink strategy appears to be relevant

Reviewer: Typo:
```suggestion
* Pre-training in a self-supervised way using a masking strategy appears to be relevant
```

> * Exploiting embedings at different levels of abstraction for images and text would seem to be a good approach

Reviewer: Typo:
```suggestion
* Exploiting embeddings at different levels of abstraction for images and text would seem to be a good approach
```
